src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -538,28 +538,28 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
         if (r > 0)
                 return 0;
 
-        /* If /boot/ is present, unused, and empty, we'll take that.
-         * Otherwise, if /efi/ is unused and empty (or missing), we'll take that.
+        /* Only mount to /efi/ if we have a XBOOTLDR partition or /efi/ already exists.
+         * Otherwise, mount to /boot/ if unused and empty (or missing) or if /efi/ is busy.
          * Otherwise, we do nothing. */
```

Comment:
Well, this means we'd no longer create `/efi/` even if `/boot/` is in use (not by XBOOTLDR)? That sounds problematic to me. We should always fall back to `/efi/`.